tests/Mockery/MockingNullableMethodsTest.php

```diff
@@ -211,4 +211,12 @@ public function it_allows_returning_null_for_nullable_int_return_types()
 
         $this->assertEquals(null, $double->nullableInt());
     }
+
+    /** @test */
+    public function it_returns_null_on_calls_to_ignored_methods_of_spies_if_return_type_is_nullable()
+    {
+        $double = \Mockery::spy(MethodWithNullableReturnType::class);
+
+        $this->assertEquals(null, $double->nullableClass());
+    }
```

Comment:
I did not add multiple tests as it looked overkill to me, but if you prefer I can add one for every specific return type contained in `MethodWithNullableReturnType`